homeassistant/components/repetier/__init__.py

```diff
@@ -37,28 +37,38 @@
 TEMP_DATA = {"tempset": "temp_set", "tempread": "state", "output": "output"}
 
 
-API_PRINTER_METHODS = {
-    "bed_temperature": {
-        "offline": {"heatedbeds": None, "state": "off"},
-        "state": {"heatedbeds": "temp_data"},
-        "temp_data": TEMP_DATA,
-        "attribute": "heatedbeds",
-    },
-    "extruder_temperature": {
-        "offline": {"extruder": None, "state": "off"},
-        "state": {"extruder": "temp_data"},
-        "temp_data": TEMP_DATA,
-        "attribute": "extruder",
-    },
-    "chamber_temperature": {
-        "offline": {"heatedchambers": None, "state": "off"},
-        "state": {"heatedchambers": "temp_data"},
-        "temp_data": TEMP_DATA,
-        "attribute": "heatedchambers",
-    },
-    "current_state": {
-        "offline": {"state": None},
-        "state": {
+@dataclass
+class APIMethods:
+    """API methods for properties."""
+
+    offline: dict[str, str | None]
+    state: dict[str, str]
+    temp_data: dict[str, str] | None = None
+    attribute: str | None = None
+
+
+API_PRINTER_METHODS: dict[str, APIMethods] = {
```

Comment:
Is this type annotation needed? Can't the type checking infer this?

Reply:
You're right, this can be inferred. However, without it you won't immediately get a warning if you keep an item with the old style by accident, i.e. `dict[str, APIMethods | dict]`. Thus, I like to add it.